main.go

```diff
@@ -52,7 +53,45 @@ func main() {
 		log.Fatal(errors.AddContext(err, "failed to build the api"))
 	}
 
-	logger.Printf("Starting Pinner service.\nGitRevision: %v (built %v)\n", build.GitRevision, build.BuildTime)
+	logger.Print("Starting Pinner service")
+	logger.Printf("GitRevision: %v (built %v)", build.GitRevision, build.BuildTime)
 	err = server.ListenAndServe(4000)
 	log.Fatal(errors.Compose(err, scanner.Close()))
 }
+
+// newLogger creates a new logger that can write to disk.
+//
+// The function also returns a closer function that should be called when we
+// stop using the logger, typically deferred in main.
+func newLogger(level, logfile string) (logger *logrus.Logger, closer func(), err error) {
+	logger = logrus.New()
+	// Parse and set log level.
+	logLevel, err := logrus.ParseLevel(level)
+	if err != nil {
+		return nil, nil, errors.AddContext(err, "invalid log level: "+level)
+	}
+	logger.SetLevel(logLevel)
+	// Open and start writing to the log file, unless we have an empty string,
```

Comment:
Empty strings are not great sentinel values. It's more common to pass an `io.Writer` to `newLogger` and then if you don't want to log you pass in `io.Discard`.

Reply:
+1 was pondering commenting on this myself but didn't want to overly review it as I wasn't assigned

Reply:
The entire reason for this function to exist is to be able to answer the question "was a log file defined?" and to be able to open a writer to it.

Would using a more explicit sentinel value (e.g. "DO_NOT_LOG_TO_FILE") make this better?

Reply:
I think the most common approach would be to add a `FILE_LOGGING_ENABLED` environment variable, perhaps in combination with a `FILE_LOGGING_PATH`. Or `LOG_ENABLED` and `LOG_PATH` or something along those lines that makes the most sense?

Reply:
Aha I see you already have `PINNER_LOG_FILE`, so maybe `PINNER_LOG_ENABLED`?

Reply:
F/U: https://github.com/SkynetLabs/pinner/issues/39